src/gravel/controllers/resources/network.py

```diff
@@ -0,0 +1,305 @@
+
+from gravel.controllers.gstate import Ticker
+from gravel.controllers.utils import aqr_run_cmd
+
+logger: logging.Logger = fastapi_logger
+
+
+class InterfaceConfigModel(BaseModel):
+    # Simplified "UI friendly" version of the actual config, i.e. we're not
+    # exposing everything in /etc/sysconfig/network/ifcfg-*, only the bits that
+    # we want the user to be able to change.
+    # TODO: bootproto would be better as a StrEnum (so we could force "dhcp" or "static")
+    bootproto: str = Field("dhcp", title="Boot Protocol")
+    ipaddr: Optional[str] = Field("", title="IP Address")
+    # If bonding_slaves is _not_ empty, this interface is a bonding master.
+    bonding_slaves: Optional[List] = Field(
+        [], title="Bonding slave network interfaces"
+    )
+    # TODO: Expose bonding module options
+
+
+# This is actually useful to have, because the interface might not *have* any config yet,
+# and in future we might want to see statistics, state (up/down, etc.) which would all
+# belong in this model.  OTOH, if we can get stats from cephadm, maybe this should be
+# dropped and we just do a dict of InterfaceConfigModels instead, and drop this extra
+# layer.
+class InterfaceModel(BaseModel):
+    name: str = Field(title="Logical Name")
+    config: Optional[InterfaceConfigModel] = Field(
+        None, title="Interface config"
+    )
+    # TODO: could add stats, current state etc. here but note that this would
+    # duplicate information available via /api/local/nodeinfo (which comes from cephadm)
+
+
+class RouteModel(BaseModel):
+    destination: str = Field("default", title="Destination")
+    gateway: str = Field("-", title="Gateway")
+    interface: str = Field("-", title="Interface")
+
+
+class NetworkConfigModel(BaseModel):
+    interfaces: Dict[str, InterfaceModel]
+    nameservers: List[str]
+    routes: List[RouteModel]
+
+
+class Network(Ticker):
+
+    _interfaces: Dict[str, InterfaceModel] = {}
+    _nameservers: List[str] = []
+    _routes: List[RouteModel] = []
+    _is_busy: bool = False
+
+    def __init__(self, interval: float) -> None:
+        super().__init__(interval)
+
+    @property
+    def interfaces(self) -> Dict[str, InterfaceModel]:
+        return self._interfaces
+
+    @property
+    def nameservers(self) -> List[str]:
+        return self._nameservers
+
+    @property
+    def routes(self) -> List[RouteModel]:
+        return self._routes
+
+    async def _do_tick(self) -> None:
+        self._refresh_interfaces()
+        self._refresh_nameservers()
+        self._refresh_routes()
+        logger.debug(self._interfaces)
+        logger.debug(self._nameservers)
+        logger.debug(self._routes)
+
+    async def _should_tick(self) -> bool:
+        return not self._is_busy
+
+    def _refresh_interfaces(self) -> None:
+        # interfaces is the union of the physical interfaces (/sys/class/net/...) and
+        # any interfaces for which we have config files (/etc/sysconfig/network/ifcfg-*)
```

Comment:
Is it correct to assume the config file in that position will be up to date? Would we be better off using `ip addr` and associated utils to get the information? Using the latter would also likely make it easier to move between wicked and networkmanager.